src/Raven.Server/Documents/PeriodicBackup/GoogleCloudStorage/RavenGoogleCloudStorageClient.cs

```diff
@@ -0,0 +1,158 @@
+        {
+            try
+            {
+                _client = StorageClient.Create(GoogleCredential.FromJson(jsonCredential));
+            }
+            catch (Exception e)
+            {
+                throw new ArgumentException("Wrong format for account key.", e);
+            }
+
+            if (JObject.Parse(jsonCredential).TryGetValue(ProjectIdPropertyName, StringComparison.InvariantCultureIgnoreCase, out var value))
+            {
+                _projectId = value.Value<string>();
+            }
+
+            _bucketName = bucketName;
+            CancellationToken = cancellationToken ?? CancellationToken.None;
+
+            _progress = progress;
+        }
+
+        public async Task<Object> UploadObjectAsync(string fileName, Stream stream, Dictionary<string, string> metadata = null)
+        {
+
+            return await _client.UploadObjectAsync(
+                new Object
+                {
+                    Bucket = _bucketName,
+                    Name = fileName,
+                    ContentType = "application/octet-stream",
+                    Metadata = metadata
+                }, stream,
+                cancellationToken: CancellationToken,
+                progress: new Progress<IUploadProgress>(p =>
+                {
+                    if (_progress == null)
+                        return;
+
+                    switch (p.Status)
+                    {
+                        case UploadStatus.Starting:
+                        case UploadStatus.NotStarted:
+                            _progress.UploadProgress.ChangeState(UploadState.PendingUpload);
+                            break;
+                        case UploadStatus.Completed:
+                            _progress.UploadProgress.ChangeState(UploadState.Done);
+                            break;
+                        case UploadStatus.Uploading:
+                            _progress.UploadProgress.ChangeState(UploadState.Uploading);
+                            break;
+                    }
+
+                    _progress.UploadProgress.UploadedInBytes = p.BytesSent;
+                    _progress.OnUploadProgress();
+                }));
+        }
+
+        public async Task DownloadObjectAsync(string fileName, Stream stream)
+        {
+            await _client.DownloadObjectAsync(
+                _bucketName,
+                fileName,
+                cancellationToken: CancellationToken,
+                destination: stream
+            ).ConfigureAwait(false);
+        }
+
+        public async Task<Object> GetObjectAsync(string fileName)
+        {
+            return await _client.GetObjectAsync(
+                _bucketName,
+                fileName,
+                cancellationToken: CancellationToken
+            ).ConfigureAwait(false);
+        }
+
+        public async Task DeleteObjectAsync(string fileName)
+        {
+            await _client.DeleteObjectAsync(
```

Comment:
why awaiting?

Reply:
Removed & change to void.

Reply:
We need to wait before starting a new test